runtime/android/core_internal/src/org/xwalk/core/internal/SwipeRefreshHandler.java

```diff
@@ -0,0 +1,185 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Copyright (c) 2016 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.xwalk.core.internal;
+
+import android.content.Context;
+import android.view.ViewGroup.LayoutParams;
+
+import org.chromium.base.annotations.JNINamespace;
+import org.chromium.base.metrics.RecordUserAction;
+import org.chromium.base.TraceEvent;
+import org.chromium.content.browser.ContentViewCore;
+import org.chromium.content.browser.OverscrollRefreshHandler;
+import org.chromium.third_party.android.swiperefresh.SwipeRefreshLayout;
+
+import org.xwalk.core.internal.R;
+
+/**
+ * An overscroll handler implemented in terms a modified version of the Android
+ * compat library's SwipeRefreshLayout effect.
+ */
+@JNINamespace("content")
```

Comment:
It's best to keep it  aligned with the original copy. I do not think it is good to introduce any new diff.